config/environments/test.rb

```diff
@@ -56,6 +56,12 @@
 
   config.i18n.default_locale = :en
   config.i18n.fallbacks = true
+
+  config.to_prepare do
+    # Force Status to always be SHAPE_TOO_COMPLEX
+    # Ref: https://github.com/mastodon/mastodon/issues/23644
+    10.times { |i| Status.allocate.instance_variable_set(:"@ivar_#{i}", nil) }
+  end
```

Comment:
Oh my bad, I forgot I had to modify the code a bit to cause the repro:

```diff
diff --git a/app/controllers/api/v1/timelines/home_controller.rb b/app/controllers/api/v1/timelines/home_controller.rb
index ae6dbcb8b..4859c7cdc 100644
--- a/app/controllers/api/v1/timelines/home_controller.rb
+++ b/app/controllers/api/v1/timelines/home_controller.rb
@@ -22,6 +22,7 @@ class Api::V1::Timelines::HomeController < Api::BaseController
 
   def cached_home_statuses
     cache_collection home_statuses, Status
+    cache_collection home_statuses, Status
   end
 
   def home_statuses
```

It's because the controller spec always start with an empty cache, we'd need to warm the cache first before making the request. Shouldn't be too hard.